src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Windows.cs

```diff
@@ -679,6 +669,18 @@ private static unsafe ValueTask WriteFileGatherAsync(SafeFileHandle handle, Memo
             return new ValueTask(vts, vts.Version);
         }
 
+        private static async ValueTask WriteGatherAtOffsetMultipleSyscallsAsync(SafeFileHandle handle, IReadOnlyList<ReadOnlyMemory<byte>> buffers, long fileOffset, CancellationToken cancellationToken)
+        {
+            long bytesWritten = 0;
+            int buffersCount = buffers.Count;
+            for (int i = 0; i < buffersCount; i++)
+            {
+                ReadOnlyMemory<byte> rom = buffers[i];
+                await WriteAtOffsetAsync(handle, rom, fileOffset + bytesWritten, cancellationToken).ConfigureAwait(false);
+                bytesWritten += rom.Length;
```

Comment:
We shouldn't need bytesWritten anymore... this can just update fileOffset.

Reply:
Fixed.